### PR TITLE
[#75] [KMM] [Swift] [Integration] As a user, I can see localized text

### LINKED ISSUE #75: [KMM] [Swift] [Integration] As a user, I can see localized text
## Why

User should be able to see the app in their language, if supported.

Localize texts should be from single source: Kotlin shared module.
 
## Acceptance Criteria

- Create a localized file in `Shared` module.
- Remove existing localization in iOS module.

### PR DESCRIPTION
**Pull request by blyscuit**
- close #75 

## What happened

- Add localization to `Shared` module.
- Replace localization in Swift.

## Insight

- Add [Moko resources](https://github.com/icerockdev/moko-resources) for localization.
- Replace existing localization with Moko.

### CODE REVIEW

File: iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme
Comment (minhnimble) [inline iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme]: Just curious, why do we have to introduce this change to support the new Moko library? Can you help elaborate a bit on the cause with some reference links maybe because I don't find any mention point about this issue in their [own repo](https://github.com/icerockdev/moko-resources)? I raise this because if this is a valid change, we might want to update our existing iOS template for support so having some additional information would also be helpful 😁

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt
Comment (minhnimble) [inline shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt:16]: Should we name this function to the below suggestion as it only supports `common_error`?
```suggestion
    fun localizedError(id: String?): StringDesc? {
        if (id == MR.strings.common_error.toString()) {
            return StringDesc.Resource(MR.strings.common_error)
        }
        return null
    }
```

File: shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalizeTest.kt
Comment (minhnimble) [inline shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalizeTest.kt]: I think we miss a test case for null return 🙈

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift
Comment (minhnimble) [inline iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift]: Just a thought, but maybe we can extract `LocalizeId()` to a constant for reusing. Then we can clean up the Typealias declaration for `MR.strings` for simplicity. What do you think? 🤔

File: iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme
Comment (blyscuit) [reply iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme]: The script 
```
			shellScript = "\"$SRCROOT/../gradlew\" -p \"$SRCROOT/../\" :Shared:copyFrameworkResourcesToApp \\\n    -Pmoko.resources.PLATFORM_NAME=$PLATFORM_NAME \\\n    -Pmoko.resources.CONFIGURATION=$CONFIGURATION \\\n    -Pmoko.resources.BUILT_PRODUCTS_DIR=$BUILT_PRODUCTS_DIR \\\n    -Pmoko.resources.CONTENTS_FOLDER_PATH=$CONTENTS_FOLDER_PATH\n";
```
Will try to read the files for `CONFIGURATION` at `BUILT_PRODUCTS_DIR` which have some issue with `space` in the naming, something with the MOKO script is reading the `space` character as `break` and will not be able to find the autogenerated files.

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt
Comment (blyscuit) [reply shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt:16]: This extension is allow to localized any needed id, from the id's string description. I think it's best to name it generically. Currently there's only `common_error` that needed this feature but any string that is dynamically localized from KMM module needs to have this feature.

File: iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift
Comment (blyscuit) [reply iosApp/Survey/Sources/Presentation/Modules/Login/LoginView.swift]: Updated to a singleton in https://github.com/blyscuit/kmm-swiftUI-IC/pull/76/commits/8d925c48c3c6d8ae39e13d428293a1ce42ac6e63

File: iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme
Comment (minhnimble) [reply iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme]: This looks like a scripting issue to me that the `space` needs to be converted to `"\ "` for it to work. What do you think about updating the `BUILT_PRODUCTS_DIR` to support the new space representation by using the following [approach](https://stackoverflow.com/a/51406339)? 🤔

-> Instead of using the original `BUILT_PRODUCTS_DIR`, we can convert it to `NEW_BUILT_PRODUCTS_DIR`:
```
NEW_BUILT_PRODUCTS_DIR ="${BUILT_PRODUCTS_DIR// /\\ }"
```
=> Then we can update this to: `-Pmoko.resources.BUILT_PRODUCTS_DIR=$NEW_BUILT_PRODUCTS_DIR`. Let me know what you think 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt
Comment (minhnimble) [reply shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/helpers/MokoLocalize.kt:16]: ah okay, it makes sense now 👌

File: iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme
Comment (blyscuit) [reply iosApp/Survey.xcodeproj/xcshareddata/xcschemes/Survey Staging.xcscheme]: @minhnimble Fixed with explicated string conversion. Thank you for the suggestions 🙏 
Screen Shot 2022-11-23 at 09 30 46

**Comment by minhnimble** [comment]
@blyscuit it looks like we have an CI issue with Setup Ruby step, need to fix it 🛠

Screenshot 2022-11-23 at 10 15 23

**Comment by blyscuit** [comment]
@minhnimble It is fixed [here](https://github.com/blyscuit/kmm-swiftUI-IC/pull/84). Waiting for one more approval.

**Comment by minhnimble** [comment]
ohh, then should we base this PR on top of the bugfix PR? 🤔

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

File: .github/workflows/ios_review_pull_request.yml
Comment (suho) [inline .github/workflows/ios_review_pull_request.yml:29]: Why you made this change? 60 minutes is too much 😁

File: iosApp/Survey/Sources/Supports/Extensions/MokoResources/StringResource+Localized.swift
Comment (suho) [inline iosApp/Survey/Sources/Supports/Extensions/MokoResources/StringResource+Localized.swift]: Just curious, can we have `callAsFunction` instead?
```swift
extension StringResource {

	func callAsFunction() {
		desc().localized()
	}
}

// then
String.localizeId.reset_password_button_reset()
```

File: .github/workflows/ios_review_pull_request.yml
Comment (blyscuit) [reply .github/workflows/ios_review_pull_request.yml:29]: On occasional clean builds, it can take a long time to build, with more UITest, it can take even longer. I had build failed because timeout so 60 here is for the safe side, it should never exceed 60 tho.

File: iosApp/Survey/Sources/Supports/Extensions/MokoResources/StringResource+Localized.swift
Comment (blyscuit) [reply iosApp/Survey/Sources/Supports/Extensions/MokoResources/StringResource+Localized.swift]: Nice suggestion 🚀

**Review by doannimble — APPROVED** [review]
lgtm

**Pull request merged**
